File: fast-tax-nodejs-examples/get_best_match_soap_sdk_example.js
import { GetBestMatchSoap } from '../fast-tax-nodejs/SOAP/get_best_match_soap.js';

async function GetBestMatchSoapGo(licenseKey, isLive) {
    console.log("\n---------------------------------");
    console.log("FastTax - GetBestMatch - SOAP SDK");
    console.log("---------------------------------");

    const address = "27 E Cota St";
    const address2 = "";
    const city = "Santa Barbara";
    const state = "CA";
    const zip = "93101";
    const taxType = "sales";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Address        : ${address}`);
    console.log(`Address2       : ${address2}`);
    console.log(`City           : ${city}`);
    console.log(`State          : ${state}`);
    console.log(`Zip            : ${zip}`);
    console.log(`Tax Type       : ${taxType}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const ft = new GetBestMatchSoap(
            address,
            address2,
            city,
            state,
            zip,
            taxType,
            licenseKey,
            isLive,
            timeoutSeconds
        );
        const response = await ft.invokeAsync();

        console.log("\n* Tax Info *\n");
        if (response) {
            console.log(`Match Level  : ${response.MatchLevel}`);
            if (response.TaxInfoItems && response.TaxInfoItems.BestMatchTaxInfo.length > 0) {
                for (const taxInfo of response.TaxInfoItems.BestMatchTaxInfo) {
                    console.log(`Zip              : ${taxInfo.Zip}`);
                    console.log(`City             : ${taxInfo.City}`);
                    console.log(`County           : ${taxInfo.County}`);
                    console.log(`State Abbrev     : ${taxInfo.StateAbbreviation}`);
                    console.log(`State Name       : ${taxInfo.StateName}`);
                    console.log(`Total Tax Rate   : ${taxInfo.TaxRate}`);
                    console.log(`State Rate       : ${taxInfo.StateRate}`);
                    console.log(`City Rate        : ${taxInfo.CityRate}`);
                    console.log(`County Rate      : ${taxInfo.CountyRate}`);
                    console.log(`County Dist Rate : ${taxInfo.CountyDistrictRate}`);
                    console.log(`City Dist Rate   : ${taxInfo.CityDistrictRate}`);
                    console.log(`Special Dist Rate: ${taxInfo.SpecialDistrictRate}`);
                    console.log(`Total Tax Exempt : ${taxInfo.TotalTaxExempt}`);
                    console.log(`Notes Codes      : ${taxInfo.NotesCodes}`);
                    console.log(`Notes Desc       : ${taxInfo.NotesDesc}`);

                    console.log("\n* Information Components *\n");
                    if (taxInfo.InformationComponents && taxInfo.InformationComponents?.InformationComponent.length > 0) {
                        taxInfo.InformationComponents.InformationComponent.forEach((component) => {
                            console.log(`${component.Name}: ${component.Value}`);
                        });
                    } else {
                        console.log("No information components found.");
                    }
                }
            } else {
                console.log("No tax info items found.");
            }
        } else {
            console.log("No tax info found.");
        }

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error Number  : ${response.Error.Number}`);
            console.log(`Error Location: ${response.Error.Location}`);
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetBestMatchSoapGo };

File: fast-tax-python-examples/get_best_match_soap_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../fast-tax-python/SOAP"))

from get_best_match_soap import GetBestMatchSoap


def get_best_match_soap_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n---------------------------------")
    print("FastTax - GetBestMatch - SOAP SDK")
    print("---------------------------------")

    address = "136 W Canon Perdido St Ste D"
    address2 = ""
    city = "Santa Barbara"
    state = "CA"
    zip = "93101"
    tax_type = "sales"
    timeout_seconds = 15

    print("\r\n* Input *\r\n")
    print(f"Address        : {address}")
    print(f"Address2       : {address2}")
    print(f"City           : {city}")
    print(f"State          : {state}")
    print(f"Zip            : {zip}")
    print(f"Tax Type       : {tax_type}")
    print(f"License Key    : {license_key}")
    print(f"Is Live        : {is_live}")
    print(f"Timeout Seconds: {timeout_seconds}")

    try:
        service = GetBestMatchSoap(license_key, is_live, timeout_seconds * 1000)
        response = service.get_best_match(address, address2, city, state, zip, tax_type)

        if not hasattr(response, 'Error') or not response.Error:
            print("\r\n* Tax Info *\r\n")
            print(f"Match Level      : {response.MatchLevel}")
            if hasattr(response, 'TaxInfoItems') and response.TaxInfoItems:
                items = response.TaxInfoItems.BestMatchTaxInfo if hasattr(response.TaxInfoItems, 'BestMatchTaxInfo') else []
                if not isinstance(items, list):
                    items = [items]
                for tax_info in items:
                    print(f"Zip              : {getattr(tax_info, 'Zip', None)}")
                    print(f"City             : {getattr(tax_info, 'City', None)}")
                    print(f"County           : {getattr(tax_info, 'County', None)}")
                    print(f"State Abbrev     : {getattr(tax_info, 'StateAbbreviation', None)}")
                    print(f"State Name       : {getattr(tax_info, 'StateName', None)}")
                    print(f"Total Tax Rate   : {getattr(tax_info, 'TaxRate', None)}")
                    print(f"State Rate       : {getattr(tax_info, 'StateRate', None)}")
                    print(f"City Rate        : {getattr(tax_info, 'CityRate', None)}")
                    print(f"County Rate      : {getattr(tax_info, 'CountyRate', None)}")
                    print(f"County Dist Rate : {getattr(tax_info, 'CountyDistrictRate', None)}")
                    print(f"City Dist Rate   : {getattr(tax_info, 'CityDistrictRate', None)}")
                    print(f"Special Dist Rate: {getattr(tax_info, 'SpecialDistrictRate', None)}")
                    print(f"Total Tax Exempt : {getattr(tax_info, 'TotalTaxExempt', None)}")
                    print(f"Notes Codes      : {getattr(tax_info, 'NotesCodes', None)}")
                    print(f"Notes Desc       : {getattr(tax_info, 'NotesDesc', None)}")

                    print("\r\n* Information Components *\r\n")
                    if hasattr(tax_info, 'InformationComponents') and tax_info.InformationComponents:
                        components = tax_info.InformationComponents.InformationComponent
                        if not isinstance(components, list):
                            components = [components]
                        for component in components:
                            print(f"{getattr(component, 'Name', None)}: {getattr(component, 'Value', None)}")
                    else:
                        print("No information components found.")
            else:
                print("No tax info items found.")
        else:
            print("No tax info found.")

        if hasattr(response, 'Error') and response.Error:
            print("\r\n* Error *\r\n")
            print(f"Error Desc    : {response.Error.Desc}")
            print(f"Error Number  : {response.Error.Number}")
            print(f"Error Location: {response.Error.Location}")

    except Exception as e:
        print("\r\n* Error *\r\n")
        print(f"Exception occurred: {str(e)}")
